File: elixir/lasagna/lib/lasagna.ex
defmodule Lasagna do
  def expected_minutes_in_oven do
    40
  end

  def remaining_minutes_in_oven(elapsed) do
    expected_minutes_in_oven - elapsed
  end

  def preparation_time_in_minutes(layers) do
    2 * layers
  end

  def total_time_in_minutes(layers, elapsed) do
    preparation_time_in_minutes(layers) + elapsed
  end

  def alarm do
    "Ding!"
  end
end

File: go/lasagna/lasagna.go
package lasagna

const OvenTime = 40

func RemainingOvenTime(elapsed int) int {
	return OvenTime - elapsed
}

func PreparationTime(layers int) int {
	return layers * 2
}

func ElapsedTime(layers, elapsed int) int {
	return PreparationTime(layers) + elapsed
}
